docs/adr/0016-auth-for-settings-redesign.md

```diff
@@ -0,0 +1,114 @@
+# Authentication in Settings Redesign
+
+- Status: proposed
+- Deciders: Les Orchard, Lauren Zugai, Ben Bangert, Ryan Kelly
+- Date: 2020-04-15
```

Comment:
I think @dannycoates should be added here. Also, the date should be `05` not `04`).

(Also not entirely sure I should be given any credit here, I don't know that I added anything meaningful to this conversation. 😛)